homeassistant/components/denon/media_player.py

```diff
@@ -111,10 +111,18 @@ def _setup_sources(self, telnet):
         if nsfrn:
             self._name = nsfrn
 
-        # SSFUN - Configured sources with names
+        # SSFUN - Configured sources with (optional) names
         self._source_list = {}
         for line in self.telnet_request(telnet, "SSFUN ?", all_lines=True):
-            source, configured_name = line[len("SSFUN") :].split(" ", 1)
+            ssfun = line[len("SSFUN") :].split(" ", 1)
+
+            source = ssfun[0]
+            if len(ssfun) > 1:
```

Comment:
To ensure that the list has only two elements and that the second element has a value.
```suggestion
            if len(ssfun) == 2 and ssfun[1]:
```

Reply:
Sure, thanks.
I tried that with my own setup and it seems to work fine.